Loop/Managers/Missed Meal Detection/MealDetectionManager.swift

```diff
@@ -0,0 +1,311 @@
+                        return item
+                    }
+
+                    return GlucoseEffect(startDate: effect.startDate.dateFlooredToTimeInterval(delta),
+                                         quantity: effect.quantity)
+                }
+            
+            for effect in processedICE {
+                let value = effect.quantity.doubleValue(for: unit)
+                effectValueCache[effect.startDate] = (effectValueCache[effect.startDate] ?? 0) + value
+            }
+            
+            var unexpectedDeviation: Double = 0
+            var mealTime = now
+            
+            /// Dates the algorithm uses when computing effects
+            /// Have the range go from newest -> oldest time
+            let summationRange = LoopMath.simulationDateRange(from: intervalStart,
+                                                    to: now,
+                                                    delta: delta)
+                                          .reversed()
+            
+            /// Dates the algorithm is allowed to check for the presence of a missed meal
+            let dateSearchRange = Set(LoopMath.simulationDateRange(from: intervalStart,
+                                                         to: intervalEnd,
+                                                         delta: delta))
+            
+            /// Timeline used for debug purposes
+            var missedMealTimeline: [(date: Date, unexpectedDeviation: Double?, mealThreshold: Double?, rateOfChangeThreshold: Double?)] = []
+            
+            for pastTime in summationRange {
+                guard
+                    let unexpectedEffect = effectValueCache[pastTime],
+                    !carbEntries.contains(where: { $0.startDate >= pastTime })
+                else {
+                    missedMealTimeline.append((pastTime, nil, nil, nil))
+                    continue
+                }
+                
+                unexpectedDeviation += unexpectedEffect
+
+                guard dateSearchRange.contains(pastTime) else {
+                    /// This time is too recent to check for a missed meal
+                    missedMealTimeline.append((pastTime, unexpectedDeviation, nil, nil))
+                    continue
+                }
+                
+                /// Find the threshold based on a minimum of `missedMealGlucoseRiseThreshold` of change per minute
+                let minutesAgo = now.timeIntervalSince(pastTime).minutes
+                let deviationChangeThreshold = MissedMealSettings.glucoseRiseThreshold * minutesAgo
+                
+                /// Find the total effect we'd expect to see for a meal with `carbThreshold`-worth of carbs that started at `pastTime`
+                guard let modeledMealEffectThreshold = self.effectThreshold(mealStart: pastTime, carbsInGrams: MissedMealSettings.minCarbThreshold) else {
```

Comment:
Would it be simpler to just lookup CSF based on CR and ISF for `pastTime`, and use that to turn minCarbThreshold into an expected glucose change?

Reply:
I thought about that and my concern with that approach was the potential for subtle bugs with in terms of handling overrides - I generally tried to use LoopKit's effects functions because then this is handled automatically.

Reply:
LDM does have access to the schedules with recent overrides applied. I would accept this PR with a dependency on CarbStore, but if you're up for it, it does seem a little cleaner, and maybe easier to test and maintain, to keep it free of that dependency.

Reply:
I'm definitely up for it - I want to ensure this is maintainable 😁